### PR TITLE
Add activity listener that consumes lifecycle events and feeds AppStartupTraceEmitter

### PR DESCRIPTION
**Pull request by bidetofevil**
## Goal

Add an activity lifecycle callback handler funnel application lifecycle events to the `AppStartupTraceEmitter`. The implementation to observe the frame render event is based on PY's blog post here: https://blog.p-y.wtf/tracking-android-app-launch-in-production

Almost all of the complexity is in that code, and while I don't have automation yet, I have tested this manually in API 21 and 34, and many in between. That said, this is the part that needs the most scrutiny to determine whether this is viable to introduce now

## Testing

Manual testing done

### CODE REVIEW

**Comment by bidetofevil** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/501?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#501** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/501?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#500** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/500?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#499** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/499?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#468** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/468?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#467** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/467?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bidetofevil and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/startup/StartupTracker.kt [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.view.View
+import android.view.ViewTreeObserver
+import android.view.Window
+import io.embrace.android.embracesdk.annotation.StartupActivity
+import io.embrace.android.embracesdk.internal.utils.VersionChecker
+
+/**
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/startup/StartupTracker.kt:15]: I'm going to rewrite the comments tomorrow....

**Review by fractalwrench — APPROVED** [review]
LGTM

**Comment by bidetofevil** [comment]
Commits merged as part of another PR

**Pull request closed without merging**